src/context/AppProvider/AppContext.tsx

```diff
@@ -154,10 +155,9 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
   }, [dispatch, wallet, supportedChains, isSnapInstalled])
 
   // once portfolio is done loading, fetch fox voting power
-  useEffect(() => {
-    if (portfolioLoadingStatus === 'loading') return
-    dispatch(snapshotApi.endpoints.getVotingPower.initiate(requestedAccountIds))
-  }, [dispatch, requestedAccountIds, portfolioLoadingStatus])
+  useGetVotingPowerQuery(undefined, {
+    skip: !isFoxDiscountsEnabled || portfolioLoadingStatus === 'loading',
+  })
```

Comment:
is this still required as the ui also has its own instance of this query?

Reply:
Strictly speaking this isn't required anymore, though I would be inclined to keep it here as the query can be very, very long to run (will be mitigated as a follow-up by persisting the snapshots API so we can optimistically get the voting power until we get fresh data).This will ensure the data is ready earlier (unless they access the app from swapper page directly or click there directly after opening the app), at the expense of making more requests for users not swapping